### PR TITLE
Initial set of realtime tests

### PR DESCRIPTION
**Pull request by indeyets**
1. backported some realtime optimizations from `postgres` branch
2. added testing framework for Socket.io events
3. added some tests for deliverability of posts in realtime
4. fixed issue, which allowed to deliver posts to banned users

### CODE REVIEW

File: app/pubsub-listener.js [hunk 1]
```diff
@@ -143,10 +143,21 @@ export default class PubsubListener {
...
+        return;
+      }
 
-      if (valid)
-        socket.emit(type, json)
+      const authorBans = await dbAdapter.getUserBansIds(post.userId)
+
+      if (authorBans.indexOf(user.id) >= 0) {
+        return;
+      }
```
Comment (clbn) [inline app/pubsub-listener.js:160]: For the sake of clarity, can we move this piece to some subroutine, similar to `.isBannedFor`? Or maybe we have it somewhere already?

File: app/pubsub-listener.js [hunk 2]
```diff
@@ -147,7 +147,9 @@ export default class PubsubListener {
         return;
       }
 
-      if (await post.isBannedFor(user.id)) {
+      const banIds = await user.getBanIds()
+
+      if (banIds.indexOf(post.userId) >= 0) {
```
Comment (clbn) [inline app/pubsub-listener.js:152]: Isn't it equivalent?

**Comment by clbn** [comment]
I left a couple of in-code comments. Also, it seems you're using `;` in the new code. It sort of contradicts current code style, doesn't it?

File: app/pubsub-listener.js (hunk 2 above)
Comment (indeyets) [inline app/pubsub-listener.js:152]: it is. but the idea is, that I can reuse `banIds` a bit later in this code, while `isBannedFor` hid the data from me

File: app/pubsub-listener.js (hunk 1 above)
Comment (indeyets) [inline app/pubsub-listener.js:160]: usage of `isBannedFor` is gone in a next commit anyway

**Comment by indeyets** [comment]
@clbn regarding semicolons: wasn't it your plan to introduce them? I can not do it in a single commit right now (as that would seriously complicate merge of `postgres` branch), but I can write new code using new style.

the plan is to introduce extensive eslint-ruleset as soon as merge happens

**Comment by clbn** [comment]
@indeyets On semicolons: AFAIR, at that meeting I said something like: "Yes, I can do it, but it's probably up to Indeyets to decide since he's its maintainer". So sure, let's introduce the big change + eslint later then.

So at this point, this change looks good to me.

**Pull request merged**